### PR TITLE
[tests] Properly test for multiple headers with same name in probes

### PR DESCRIPTION
**Pull request by TooTallNate**
Updates the `responseHeaders` probe checks to properly test for multiple headers with the same name.

Previously the probes were using `headers.get()` which concats multiple headers into a single string, which results in the test not really checking if there are in fact multiple headers with the same name. Using `headers.raw()` allows us to properly test for that.

A couple of Python tests that were already checking for multiple `set-cookie` headers needed to be updated to match the full value, since the check now properly validates the full string match of each header (before it was basically just doing a `string.includes()` check).

This is a precursor for https://github.com/vercel/vercel/pull/9533.

### CODE REVIEW

**Review by Ethan-Arrowood — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

File: test/lib/deployment/test-deployment.js [hunk 1]
```diff
@@ -220,26 +220,47 @@ async function runProbe(probe, deploymentId, deploymentUrl, ctx) {
...
-        : typeof expected === 'string' &&
-          expected.startsWith('/') &&
-          expected.endsWith('/')
-        ? new RegExp(expected.slice(1, -1)).test(actual)
-        : expected === actual;
-      if (!isEqual) {
-        const headers = Array.from(resp.headers.entries())
-          .map(([k, v]) => `  ${k}=${v}`)
-          .join('\n');
+      const actualArr = rawHeaders[header.toLowerCase()];
```
Comment (styfle) [inline test/lib/deployment/test-deployment.js:230]: Are raw headers always lowercase keys?

**Review by styfle — APPROVED** [review]

File: test/lib/deployment/test-deployment.js (hunk 1 above)
Comment (TooTallNate) [reply test/lib/deployment/test-deployment.js:230]: They are not.

Comment (styfle) [reply test/lib/deployment/test-deployment.js:230]: So why is the `header.toLowerCase()` there? Are we assuming the server will always lowercase the key before sending the response?

Comment (TooTallNate) [reply test/lib/deployment/test-deployment.js:230]: I guess so. Some existing tests were using lowercase.

**Pull request merged**